integration/tests/cook/util.py

```diff
@@ -115,14 +115,16 @@ def impersonating(self, other_user):
         return type(self)(self.name, impersonatee=other_username)
 
     def __enter__(self):
-        logger.debug(f'Switching to user {self.name}')
         if self.impersonatee:
+            logger.debug(f'Switching to user {self.name} with impersonatee {self.impersonatee}')
             self.previous_impersonatee = session.headers.get(IMPERSONATION_HEADER)
             session.headers[IMPERSONATION_HEADER] = self.impersonatee
+        else:
+            logger.debug(f"Not switching impersonation header to {self.name} due to None self.impersonatee")
```

Comment:
I think we should just remove this `else` block.

Reply:
Ok.